ydb/core/kqp/executer_actor/kqp_data_executer.cpp

```diff
@@ -2103,7 +2108,8 @@ class TKqpDataExecuter : public TKqpExecuterBase<TKqpDataExecuter, EExecType::Da
         }
 
         // Single-shard datashard transactions are always immediate
-        ImmediateTx = (datashardTxs.size() + evWriteTxs.size() + Request.TopicOperations.GetSize() + sourceScanPartitionsCount) <= 1
+        auto topicSize = (BufferActorId) ? TxManager->GetTopicOperations().GetSize() : Request.TopicOperations.GetSize();
```

Comment:
This is a part of old execution. There is no need to support TxManager here.

Reply:
> This is a part of old execution. There is no need to support TxManager here.

DataExecuter doesn't work well without this condition (use after move or assertion failed for check !TxManager or !BufferActorId with enabled OLTP sink).